lib/helper/user_comms.rb

```diff
@@ -0,0 +1,25 @@
+module Emerald
+  class UserCommsHelper
+
+    ERROR_INCORRECT_FILE_TYPE = "Oops! Incorrect file type received.\nExpecting a lisp file e.g. example.lisp\n"
+    ERROR_INITIALISE_WITH_STRING_IO = "Error: Initialize with StringIO objects"
+
+    def initialize(stdout)
+      raise ERROR_INITIALISE_WITH_STRING_IO if stdout.nil?
+      @stdout = stdout if stdout.respond_to?(:puts)
+    end
+
+    def verify_lisp_file(input_file)
+      pattern = /[a-zA-Z0-9\-_]+.lisp/
```

Comment:
I see your point, but it was going to be the first safe guard against non-lisps being compiled. I can see the the advantage of not wanting to type the extension, but can't really see why a user would want to use a different one (Unless I wanted to allow lisps written in a .rb file).
I also was considering making it .emerald instead so I could use my own lispy files. :)